### PR TITLE
[153] Add the layout children action

### PR DESCRIPTION
**Pull request by scosta-obeo**
### [153] Add tests for layout children action
Bug: https://github.com/eclipse-sirius/sirius-desktop/issues/153

### [153] Add the provider for the layout children action
Bug: https://github.com/eclipse-sirius/sirius-desktop/issues/153

### [153] Add the layout children action in the toolbar and the menus
Note that this action does nothing at this commit. The command for the
action is to come in the next commit.

Bug: https://github.com/eclipse-sirius/sirius-desktop/issues/153

### CODE REVIEW

**Comment by lfasani** [comment]
When `Move pinned element` is checked, `Arrange children `behaves like `Arrange All`

File: plugins/org.eclipse.sirius.diagram.ui/icons/layoutChildren@2x.png
Comment (lfasani) [inline plugins/org.eclipse.sirius.diagram.ui/icons/layoutChildren@2x.png:1]: The icon seems a little blurred compared to the others (All, Selection, Linked Border Nodes)

File: plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/menu/DiagramEditorContextMenuProvider.java [hunk 1]
```diff
@@ -235,26 +215,56 @@ private void updateSelectMenu(IMenuManager menu) {
...
 
     /**
-     * Rename the Arrange menu (new name=Layout menu), move it and reorganize it.
+     * Move Layout menu and reorganize it.
      * 
      * @param menu
      *            The parent menu containing "Format" menu
      */
-    private void moveArrangeMenuForDiagramElements(IMenuManager menu) {
+    private void reorguenizeLayoutMenu(IMenuManager menu) {
```
Comment (lfasani) [inline plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/menu/DiagramEditorContextMenuProvider.java:223]: reorganize

**Review by lfasani — CHANGES_REQUESTED** [review]

**Comment by scosta-obeo** [comment]
For bug when `Move Pinned Elements` is checked, see https://github.com/eclipse-sirius/sirius-desktop/pull/222

File: plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/menu/DiagramEditorContextMenuProvider.java (hunk 1 above)
Comment (scosta-obeo) [reply plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/menu/DiagramEditorContextMenuProvider.java:223]: Done

**Review by lfasani — APPROVED** [review]

**Pull request merged**